### PR TITLE
Fix start-stop events and flags

### PR DESCRIPTION
**Pull request by tshemsedinov**
- Added event `Application.on('stopped')`
- Added flag `Application.stopped`
- Removed event `Application.on('stop')`
- Removed event `Application.on('start')`
- Renamed flag `Application.ready` to `Application.started`
- Removed flag `Application.isInitialized`

### CODE REVIEW

File: lib/core.js [hunk 1]
```diff
@@ -534,24 +534,34 @@ impress.gsInterval = () => {
...
+impress.stop = callback => {
   impress.cache.clear();
   const servers = impress.servers;
   if (!servers) {
     impress.log.warn('No servers active');
+    callback();
     return;
   }
+
+  let count = 0;
```
Comment (belochub) [inline lib/core.js]: Can we maybe use `metasync.each()` here instead of manual counting?

File: lib/core.js [hunk 2]
```diff
@@ -534,24 +534,34 @@ impress.gsInterval = () => {
...
+  const afterClose = () => {
+    count--;
+    for (const appName in impress.applications) {
+      const application = impress.applications[appName];
+      application.emit('stop');
```
Comment (belochub) [inline lib/core.js]: Do we need both `stop` and `stopped` events?

Comment (tshemsedinov) [reply lib/core.js]: 'stopped' is enough but we have 'start' and 'started' so we can remove both 'stop' and 'start' for symmetry :smile:

File: lib/core.js (hunk 1 above)
Comment (tshemsedinov) [reply lib/core.js]: `metasync.each()` iterates `Array` but we need to iterate `Object`, we can take keys but I think counter is ok here

Comment (belochub) [reply lib/core.js]: As another option, we can use `Object.values()` here with `metasync.each()`.

File: lib/core.js (hunk 2 above)
Comment (tshemsedinov) [reply lib/core.js]: Removed

File: lib/core.js [hunk 3]
```diff
@@ -534,24 +534,33 @@ impress.gsInterval = () => {
...
   if (!servers) {
     impress.log.warn('No servers active');
+    callback();
     return;
   }
+
+  let count = 0;
+  const afterClose = () => {
+    count--;
+    for (const appName in impress.applications) {
```
Comment (lundibundi) [inline lib/core.js]: We can also use `Object.values` here I think.

File: lib/core.js (hunk 1 above)
Comment (tshemsedinov) [reply lib/core.js]: Ok

**Comment by tshemsedinov** [comment]
Landed in 1f24800

**Pull request closed without merging**